bson/src/main/org/bson/types/Decimal128.java

```diff
@@ -374,6 +383,57 @@ public boolean isNaN() {
     }
 
 
+    @Override
+    public int compareTo(final Decimal128 o) {
+        if (isNaN()) {
+            return o.isNaN() ? 0 : 1;
+        }
+        if (isInfinite()) {
+            if (isNegative()) {
```

Comment:
This inner if-else block could be extracted out into a private method.

Reply:
I didn't think it would aid in clarity to pull it out.  I was hoping for a more radical simplification.

Reply:
That's fine. It would be a challenge to come up with the right method name that would provide the necessary clarity.